src/coreclr/src/jit/rationalize.cpp

```diff
@@ -700,10 +700,14 @@ Compiler::fgWalkResult Rationalizer::RewriteNode(GenTree** useEdge, Compiler::Ge
         case GT_BLK:
             // We should only see GT_BLK for TYP_STRUCT or for InitBlocks.
             assert((node->TypeGet() == TYP_STRUCT) || use.User()->OperIsInitBlkOp());
+            // Clear the `GTF_IND_ASG_LHS` flag, which overlaps with `GTF_IND_REQ_ADDR_IN_REG`.
+            node->gtFlags &= ~GTF_IND_ASG_LHS;
```

Comment:
I would recommend moving the `GT_BLK` and `GT_OBJ` cases next to the `GT_IND` cases, so the common characteristics are clearer. You might even consider extracting them all to a common method, e.g. `RewriteIndir(GenTreeIndir* indir)`.

Reply:
Thank you, fixed. I will need to do more changes there if I want to fix missed `TGT_ANYWHERE` flag, so it will be useful to have a separate method.

Reply:
Excellent - thanks!